wrapanapi/virtualcenter.py

```diff
@@ -518,11 +518,26 @@ def vm_status(self, vm_name):
         return str(self._get_vm(vm_name, force=True).runtime.powerState)
 
     def vm_creation_time(self, vm_name):
+        """Detect the vm_creation_time either via uptime if non-zero, or by last boot time
+
+        The API provides no sensible way to actually get this value. The only way in which
+        vcenter API MAY have this is by filtering through events
+
+        Return tz-naive datetime object
+        """
         vm = self._get_vm(vm_name)
-        vcenter_time_now = self.service_instance.CurrentTime()
-        vm_uptime = vm.summary.quickStats.uptimeSeconds
-        vm_delta = timedelta(seconds=int(vm_uptime))
-        return vcenter_time_now - vm_delta
+
+        filter = vim.event.EventFilterSpec(
```

Comment:
"filter" is a Python builtin function, might be an idea to either rename this to "filter_spec" or follow the Python recommendation of appending an underscore: "filter_".